pkg/repositories/api_key_repository.go

```diff
@@ -10,11 +10,13 @@ import (
 )
 
 type ApiKeyRepository interface {
-	Create(ctx context.Context, tx db.Transaction, apiKey persistence.ApiKey) (persistence.ApiKey, error)
+	Create(ctx context.Context, apiKey persistence.ApiKey) (persistence.ApiKey, error)
 	Get(ctx context.Context, id uuid.UUID) (persistence.ApiKey, error)
 	GetForKey(ctx context.Context, apiKey uuid.UUID) (persistence.ApiKey, error)
-	GetForUser(ctx context.Context, tx db.Transaction, user uuid.UUID) ([]uuid.UUID, error)
-	Delete(ctx context.Context, tx db.Transaction, ids []uuid.UUID) error
+	GetForUser(ctx context.Context, user uuid.UUID) ([]uuid.UUID, error)
+	GetForUserTx(ctx context.Context, tx db.Transaction, user uuid.UUID) ([]uuid.UUID, error)
+	Delete(ctx context.Context, ids []uuid.UUID) error
+	DeleteTx(ctx context.Context, tx db.Transaction, ids []uuid.UUID) error
```

Comment:
Same reasoning as we can now delete some API keys without deleting the associated user so we need a no-transaction version.